internal/identity/identity.go

```diff
@@ -0,0 +1,199 @@
+package identity
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	v1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type Identity interface {
+	String() string
```

Comment:
Include a private method here to differentiate Identity from any other type that might happen to implement String().